src/components/cards/challenge/_partials/Button.tsx

```diff
@@ -2,23 +2,27 @@ import { useState } from "react";
 import CloseIcon from "@/icons/close-top-right.svg";
 import Loader from "@/components/ui/Loader";
 
-interface ButtonProps {
+export interface ButtonProps {
   text: string;
   onClick: () => void;
   loading: boolean;
 }
 
 export default function Button({ text, onClick, loading }: ButtonProps) {
-  const [isTextVisible, setistextVisible] = useState(false);
+  const [isTextVisible, setIsTextVisible] = useState(false);
   return (
-    <div className="ml-auto cursor-pointer relative" onMouseEnter={() => setistextVisible(true)} onMouseLeave={() => setistextVisible(false)} onClick={onClick}>
+    <div className="ml-auto cursor-pointer relative " onMouseEnter={() => setIsTextVisible(true)} onMouseLeave={() => setIsTextVisible(false)} onClick={onClick}
+    data-testid = "button"
+    >
       <>
         {loading ? (
-          <Loader isSmallSpinner />
+          <Loader isSmallSpinner  data-testid = "loader"/>
```

Comment:
We already have test id on the loader, you can pass your own testId as a **testId** prop